### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by pszal**
- EmptyPath added to sciond, partially #390, #392 (API there needs to be revised)
- fixed bug found by @mskd96
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/403%23issuecomment-145499258%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41133488%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41133710%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41150980%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41151135%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23issuecomment-145557496%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23issuecomment-145499258%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40aznair%2C%20%40shitz%20you%20may%20want%20to%20look%20at%20this%22%2C%20%22created_at%22%3A%20%222015-10-05T11%3A20%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%22%2C%20%22created_at%22%3A%20%222015-10-05T14%3A55%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20a8bb65ff59cf8caa36634802b669911e7bd5f8b9%20endhost/sciond.py%2031%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41133710%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20we%20define%20a%20constant%20for%20the%20%27dummy%20address%27%3F%22%2C%20%22created_at%22%3A%20%222015-10-05T11%3A31%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20don%27t%20think%20so%2C%20especially%20as%20this%20API%20will%20be%20changed%20soon%20%28I%27ll%20create%20an%20issue%20soon%29.%20For%20now%20I%20put%20%60TODO%60%22%2C%20%22created_at%22%3A%20%222015-10-05T14%3A40%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20endhost/sciond.py%3AL334-346%22%7D%2C%20%22Pull%20a8bb65ff59cf8caa36634802b669911e7bd5f8b9%20endhost/sciond.py%2027%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/403%23discussion_r41133488%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20think%20you%20could%20simplify%20this%20with%5Cr%5Cn%60%60%60%5Cr%5Cnhaddr%20%3D%20self.ifid2addr.get%28fwd_if%2C%20haddr_parse%28%5C%22IPv4%5C%22%2C%20%5C%220.0.0.0%5C%22%29%29%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222015-10-05T11%3A28%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3840858%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/shitz%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%2C%20indeed%20cleaner%22%2C%20%22created_at%22%3A%20%222015-10-05T14%3A39%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20endhost/sciond.py%3AL334-346%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/403#issuecomment-145499258'>General Comment</a></b>
- <a href='https://github.com/pszalach'></a> @aznair, @shitz you may want to look at this
- <a href='https://github.com/shitz'></a> LGTM
- [ ] <a href='#crh-comment-Pull a8bb65ff59cf8caa36634802b669911e7bd5f8b9 endhost/sciond.py 27'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/403#discussion_r41133488'>File: endhost/sciond.py:L334-346</a></b>
- <a href='https://github.com/shitz'></a> I think you could simplify this with

```
haddr = self.ifid2addr.get(fwd_if, haddr_parse("IPv4", "0.0.0.0"))
```
- <a href='https://github.com/pszalach'></a> done, indeed cleaner
- [ ] <a href='#crh-comment-Pull a8bb65ff59cf8caa36634802b669911e7bd5f8b9 endhost/sciond.py 31'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/403#discussion_r41133710'>File: endhost/sciond.py:L334-346</a></b>
- <a href='https://github.com/shitz'></a> Should we define a constant for the 'dummy address'?
- <a href='https://github.com/pszalach'></a> I don't think so, especially as this API will be changed soon (I'll create an issue soon). For now I put `TODO`

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/403?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/403?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/403'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: endhost/sciond.py [hunk 1]
```diff
@@ -330,7 +334,12 @@ def _api_handle_path_request(self, packet, sender):
         for path in paths:
             raw_path = path.pack()
             # assumed IPv4 addr
-            haddr = self.ifid2addr[path.get_fwd_if()]
+            fwd_if = path.get_fwd_if()
+            if fwd_if:
```
Comment (shitz) [inline endhost/sciond.py]: I think you could simplify this with

```
haddr = self.ifid2addr.get(fwd_if, haddr_parse("IPv4", "0.0.0.0"))
```

File: endhost/sciond.py [hunk 2]
```diff
@@ -330,7 +334,12 @@ def _api_handle_path_request(self, packet, sender):
...
+                haddr = self.ifid2addr[path.get_fwd_if()]
+            else:
+                # Set dummy host addr (path is EmptyPath)
+                haddr = haddr_parse("IPv4", "0.0.0.0")
```
Comment (shitz) [inline endhost/sciond.py]: Should we define a constant for the 'dummy address'?

File: endhost/sciond.py (hunk 1 above)
Comment (pszal) [inline endhost/sciond.py]: done, indeed cleaner

File: endhost/sciond.py (hunk 2 above)
Comment (pszal) [inline endhost/sciond.py]: I don't think so, especially as this API will be changed soon (I'll create an issue soon). For now I put `TODO`

**Pull request merged**